test/test_upload_mock.py

```diff
@@ -138,15 +162,15 @@ def test_upload_sketch(
         "ARDUINO_SKETCHBOOK_DIR": session_data_dir,
     }
 
-    if package_index:
+    # Install everything just once
+    if not os.path.isdir(session_data_dir + "/packages"):
```

Comment:
Am not sure this works on Windows. :thinking: 

I'd change it to `if not Path(session_data_dir, "packages").is_dir():`.

Reply:
it works in Windows, but your suggestion is better 👍🏼